pkg/api/mock/matcher_object.go

```diff
@@ -0,0 +1,50 @@
+package mock
+
+import (
+	"fmt"
+	"reflect"
+
+	gomegaTypes "github.com/onsi/gomega/types"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	"go.anx.io/go-anxcloud/pkg/utils/object/compare"
+)
+
+// ObjectMatcher matches mockAPIObject with provided Object
+// It compares Objects by matchFields
+func ObjectMatcher(obj types.Object, matchFields ...string) gomegaTypes.GomegaMatcher {
+	return &objectMatcher{
+		obj:         obj,
+		matchFields: matchFields,
+	}
+}
+
+type objectMatcher struct {
+	obj         types.Object
+	matchFields []string
+}
+
+func (m *objectMatcher) Match(actual interface{}) (bool, error) {
+	obj, ok := actual.(*mockAPIObject)
+	if !ok {
+		return false, fmt.Errorf("ObjectMatcher matcher expects an *mockAPIObject")
```

Comment:
on-the-fly error again. and again `fmt.Errorf` without any placeholders in your format string ^^'